src/solver/conv_MP_bidirectional_winograd.cpp

```diff
@@ -0,0 +1,536 @@
+    GenerateClangDefsym((options), "xformx_f_size", WinoFilterW);                                  \
+    GenerateClangDefsym((options), "xformy_f_size", WinoFilterH);                                  \
+    GenerateClangDefsym((options), "fdilation_w", params.kernel_stride_w);                         \
+    GenerateClangDefsym((options), "fdilation_h", params.kernel_stride_h);
+
+static inline size_t Ceil(const size_t v, const size_t m)
+{
+    assert(m > 0);
+    return (v + m - 1) / m;
+}
+#endif
+
+template <int WinoDataH, int WinoFilterH, int WinoDataW, int WinoFilterW>
+WinogradBufferInfo<WinoDataH, WinoFilterH, WinoDataW, WinoFilterW>
+GetWinoBuffer(const ConvolutionContext& params, const ConvWinoBuffType buff_type)
+{
+    DEFINE_GETXFORMHWSIZE(params)
+    DEFINE_SHADER_ALIASES(params)
+
+    WinogradBufferInfo<WinoDataH, WinoFilterH, WinoDataW, WinoFilterW> Transform_info(
+        N,
+        K,
+        C,
+        group_cnt,
+        out_H,
+        out_W,
+        R,
+        S,
+        (MemLayout_t::GCNHW),
+        (ConvWinoXformType::N_GXhXw_C_Th_Tw),
+        1,
+        GetTypeSize(params.in_data_type),
+        buff_type,
+        wino_xform_h,
+        wino_xform_w);
+
+    (void)H;
+    (void)W;
+    return Transform_info;
+}
+
+template <int WinoDataH, int WinoFilterH, int WinoDataW, int WinoFilterW>
+bool ConvMPBidirectWinograd<WinoDataH, WinoFilterH, WinoDataW, WinoFilterW>::IsApplicable(
+    const ConvolutionContext& params) const
+{
+// HIP backend required for sending ptr (buffer + offset)
+// ROCBLAS for GEMM step
+
+#if(MIOPEN_BACKEND_HIP && MIOPEN_USE_ROCBLAS)
+    static const int wino_data_tile   = std::max(WinoDataH, WinoDataW);
+    static const int wino_filter_tile = std::max(WinoFilterH, WinoFilterW);
+
+    const std::string name = params.GetStream().GetDeviceName();
+
+    if(wino_data_tile == 6 && wino_filter_tile == 3)
+        if(miopen::IsDisabled(MIOPEN_DEBUG_AMD_WINOGRAD_MPASS_F6X3{}))
+            return false;
+    if(wino_data_tile == 5 && wino_filter_tile == 3)
+        if(miopen::IsDisabled(MIOPEN_DEBUG_AMD_WINOGRAD_MPASS_F5X3{}))
+            return false;
+    if(wino_data_tile == 4 && wino_filter_tile == 3)
+        if(miopen::IsDisabled(MIOPEN_DEBUG_AMD_WINOGRAD_MPASS_F4X3{}))
+            return false;
+    if(wino_data_tile == 3 && wino_filter_tile == 3)
+        if(miopen::IsDisabled(MIOPEN_DEBUG_AMD_WINOGRAD_MPASS_F3X3{}))
+            return false;
+    if(wino_data_tile == 2 && wino_filter_tile == 3)
+        if(miopen::IsDisabled(MIOPEN_DEBUG_AMD_WINOGRAD_MPASS_F2X3{}))
+            return false;
+    if(!params.use_asm_kernels)
+        return false;
+    if(!params.rmv.IsV3())
+        return false;
+    if(!params.Is2d())
+        return false;
+    if(params.direction.IsBackwardWrW())
+        return false;
+    if(!params.IsFp32())
+        return false;
+
+    if(!(StartsWith(name, "gfx8") || StartsWith(name, "gfx9")))
```

Comment:
gfx8 was never tested with this kernel. It shouldn't be functional due to a gfx9 changes and I'm not sure it will ever assemble. Please do at least sanity checks with available hardware to check asic support.

Reply:
+1 We should't add new kernels for gfx8. Please see https://github.com/AMDComputeLibraries/MLOpen/issues/1934 for more info.

Reply:
@atamazov gfx8 removed 13 days ago